### ISSUE TITLE
Squiz.PHP.DisallowComparisonAssignment false positive when comparison inside closure

### ISSUE THREAD

**Issue by stealeks** [squizlabs/PHP_CodeSniffer#3165]
**Describe the bug**
`DisallowComparisonAssignment` triggered on callbacks with comparison inside, assigned to variables

**Code sample**
```php
<?php

$array = [];

$first = array_filter($array, function ($value) {
    if ($value > 10) {
        return false;
    }

    return true;
});

$callback = function ($value) {
    if ($value > 10) {
        return false;
    }

    return true;
};

$second = array_filter($array, $callback);

```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset name="test" xsd="vendor/squizlabs/php_codesniffer/phpcs.xsd">
    <rule ref="Squiz.PHP.DisallowComparisonAssignment"/>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above
2. Create a file called `ruleset.xml` with the code ruleset above
3. Run `phpcs --standard="ruleset.xml" test.php`
4. See error message displayed
```
FILE: test.php
---------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
---------------------------------------------------------------------------
 13 | ERROR | The value of a comparison must not be assigned to a variable
---------------------------------------------------------------------------

Time: 21ms; Memory: 4MB
```

**Expected behavior**
No output (no errors)

**Versions:**
 - OS: Linux (Ubuntu 20.04.1 LTS)
 - PHP: 7.4.3
 - PHPCS: 3.5.8
 - Standard: Custom

**Additional context**
Similar bug was already fixed in [3.5.7](https://github.com/squizlabs/PHP_CodeSniffer/releases/tag/3.5.7): 
`Fixed Squiz.PHP.DisallowComparisonAssignment false positive for methods called on an object`

**Comment by gsherwood** [comment]
Thanks for the bug report. I've committed a fix, which will be in the 3.6.0 release.
